### PR TITLE
Improve machine-controller down alert only if clusters running

### PR DESCRIPTION
**Pull request by metalmatze**
We only want to alert on machine controllers being down if their clusters are actually up and running

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: config/monitoring/prometheus/rules/machine-controller.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ groups:
 - name: machine-controller
   rules:
   - alert: MachineControllerDown
-    expr: absent(up{job="machine-controller"} == 1) and (sum(kubermatic_cluster_controller_clusters) > 0)
+    expr: absent(up{job="machine-controller"} == 1) and (kubermatic_cluster_controller_cluster_status_phase{phase="running"} > 0)
```
Comment (p0lyn0mial) [inline config/monitoring/prometheus/rules/machine-controller.yaml:5]: previously we were taking into account clusters that had been in broken state ?

Comment (mrIncompetent) [inline config/monitoring/prometheus/rules/machine-controller.yaml:5]: Phase is something which will be removed as handling the field is a huge pain & has a very high chance of being incorrect.

Comment (metalmatze) [reply config/monitoring/prometheus/rules/machine-controller.yaml:5]: Yep. Right now we are only looking for cluster CRDs existing basically. And if we then can't find a machine-controller we alert.

Comment (metalmatze) [reply config/monitoring/prometheus/rules/machine-controller.yaml:5]: How would you improve this then from a Kubermatic Controller perspective? We only want to send out machine controller down alerts if clusters are actually running. This is what this PR is about.

Comment (p0lyn0mial) [reply config/monitoring/prometheus/rules/machine-controller.yaml:5]: how about leaving it as it was ? I think that "if the cluster object exists there should also be corresponding machine-controller instance" assumption is reasonable. 

If the controller doesn't exist that might indicate that someone is playing aroud or the cluster is in inconsistent state which deserves attention.

Comment (metalmatze) [reply config/monitoring/prometheus/rules/machine-controller.yaml:5]: @p0lyn0mial Yes, but that should be a different alert and not be of concern for the MachineController being down.

**Review by mrIncompetent — APPROVED** [review]
As we have not removed the phase yet, i'm fine with merging.
When removing the phase we might want to think about adding the deletionTimestamp as a label to the cluster-metric

**Pull request merged**